scrapy/responsetypes.py

```diff
@@ -86,34 +92,33 @@ def from_filename(self, filename):
         else:
             return Response
 
-    def from_body(self, body):
-        """Try to guess the appropriate response based on the body content.
-        This method is a bit magic and could be improved in the future, but
-        it's not meant to be used except for special cases where response types
-        cannot be guess using more straightforward methods."""
-        chunk = body[:5000]
-        chunk = to_bytes(chunk)
-        if not binary_is_text(chunk):
-            return self.from_mimetype('application/octet-stream')
-        elif b"<html>" in chunk.lower():
-            return self.from_mimetype('text/html')
-        elif b"<?xml" in chunk.lower():
-            return self.from_mimetype('text/xml')
-        else:
-            return self.from_mimetype('text')
-
     def from_args(self, headers=None, url=None, filename=None, body=None):
         """Guess the most appropriate Response class based on
         the given arguments."""
         cls = Response
-        if headers is not None:
+        if cls is Response and body is not None:
```

Comment:
`cls` is always `Response` at this point.

Also, maybe we should assume the body to be `b''` if it is `None`, and pass `b''` to xtractmime as `body`.